src/engraving/infrastructure/shape.cpp

```diff
@@ -114,6 +114,28 @@ const RectF& Shape::bbox() const
     }
 }
 
+std::optional<ShapeElement> Shape::find(const std::function<bool(const ShapeElement&)>& func) const
+{
+    auto it = std::find_if(m_elements.begin(), m_elements.end(), func);
+    if (it == m_elements.end()) {
+        return std::nullopt;
+    }
+    return std::make_optional(*it);
+}
+
+std::optional<ShapeElement> Shape::find(ElementType type) const
```

Comment:
I think the implementation of this method is not entirely correct. Since the shape of a sub-item may consist of multiple ShapeElements, and all these ShapeElements are inserted into the shape of the super-item, the list of ShapeElements may contain more than one ShapeElement that belongs to the sub-item. So this method may need to return a vector of ShapeElements, or perhaps a new Shape, for convenience.
It might not yet be a problem yet since Rests for example consist of only one ShapeElement for the Rest itself, but (at least in theory) that won't always be the case. 
(@mike-spa Does this comment make sense?)